### ISSUE TITLE
[Feature Request] Flexible naming

### ISSUE THREAD

**Issue by etki** [zephir-lang/zephir#1626]
Hi!

Currently naming in configuration acts a little bit counterintuitively

### Namespace has to be specified in lower case, otherwise nothing will compile 

This may be different on windows, but linux fs is case-sensitive. This happens if i try to set my namespace to `AmaTeam` instead of `amateam`:

```
./bin/zephir build
Copying new kernel files...
Preparing for PHP compilation...
Preparing configuration file...
configure: WARNING: unrecognized options: --enable-AmaTeam
configure: WARNING: unrecognized options: --enable-AmaTeam
Compiling...
In file included from kernel/memory.h:26:0:
./php_ext.h:4:25: fatal error: php_amateam.h: No such file or directory
 #include "php_amateam.h"
                         ^
// more stuff like that
```

.c / .h files are generated in same case (`php_AmaTeam.h`), but injected as lowercase (`#include "php_amateam.h"`), and that breaks everything down. Also i'm not familiar with c / c++ development and not sure what causes `enable`-warnings.

### Source directory has to have the very same name as namespace

Just as in previous example, i have to name my root directory as `amateam` instead of `AmaTeam` just because otherwise Zephir won't find the source files. It should be trivial to find required directory ignorecasively, so end user could preserve minty structure of their folders

### Extension is named after namespace

Config has `name` section which is expected to contain the filename of target extension. However, filename is generated from namespace, which is counterintuitive and can cause naming collisions in complex projects (which build two extensions with same root namespace).

### What do i suggest?

- Add `filename` config property to give extension an explicit filename. This is backwards compatible, so no harm would be done.
- Add documentation section clarifying configuration values
- Locate directory for source files using case-sensitive then case-insensitive search. If several candidates are found, throw exception.
- Allow user to use any namespace, but lowercase every use of it when generating source files

I know all of this doesn't require any decent skill to implement and send PR, but sadly i'm totally out of time migrating some slow code to Zephir.

**Comment by betrixed** [comment]
Yes, please for this feature. Extension names and filenames should be independent of namespace root.
